src/spectral.jl

```diff
@@ -264,3 +264,25 @@ function contract(nbt::Nonbacktracking, edgespace::Vector)
     contract!(y,nbt,edgespace)
     return y
 end
+
+"""spectral_distance(G₁, G₂ [, k])
+Compute the spectral distance between undirected graphs
+G₁ and G₂ using the top k greatest eigenvalues. If k is
+ommitted, uses k = min(|G₁|,|G₂|).
+
+For further details, please refer to:
+
+JOVANOVIC, I.; STANIC, Z., 2014. Spectral Distances of
+Graphs Based on their Different Matrix Representations
+"""
+function spectral_distance(G₁::Graph, G₂::Graph, k::Integer)
+  A₁ = adjacency_matrix(G₁)
+  A₂ = adjacency_matrix(G₂)
+
+  λ₁ = k < nv(G₁)-1 ? eigs(A₁, nev=k, which=:LR)[1] : eigvals(full(A₁))[end:-1:end-(k-1)]
+  λ₂ = k < nv(G₂)-1 ? eigs(A₂, nev=k, which=:LR)[1] : eigvals(full(A₂))[end:-1:end-(k-1)]
```

Comment:
just checking, I couldn't find the answer in the article you linked: are you sure you want :LR ordering and not :LM (largest magnitude?). With :LR you are ignoring eigenvalues with large but negative real part:

```julia
julia> s=Symmetric(rand(10,10));

julia> eigs(s,which=:LR,nev=6)[1]
6-element Array{Float64,1}:
  4.71391    
  1.35498    
  1.33393    
  0.712443   
  0.374063   
 -0.000467327

julia> eigs(s,which=:LM,nev=6)[1]
6-element Array{Float64,1}:
  4.71391 
 -1.42323 
  1.35498 
  1.33393 
 -1.27153 
  0.712443
```